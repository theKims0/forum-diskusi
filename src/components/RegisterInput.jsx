import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import useInput from '../hooks/useInput';

function RegisterInput({ register }) {
  const [name, onNameChange] = useInput('');
  const [email, onEmailChange] = useInput('');
  const [password, onPasswordChange] = useInput('');

  return (
    <div className="login-content">

      <form>
        <input type="text" className="form-input" value={name} onChange={onNameChange} placeholder="Name" />
        <input type="text" className="form-input" value={email} onChange={onEmailChange} placeholder="Username" />
        <input type="password" className="form-input" value={password} onChange={onPasswordChange} placeholder="Password" />
        <button type="button" className="button-primary" onClick={() => register({ name, email, password })}>Register</button>
      </form>
      <p>
        Already have an account?
        {' '}
        <Link to="/">Login</Link>
      </p>
    </div>
  );
}

RegisterInput.propTypes = {
  register: PropTypes.func.isRequired,
};

export default RegisterInput;
