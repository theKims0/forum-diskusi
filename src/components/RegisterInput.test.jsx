/**
 * testing secenario for component register input
 * should handle name typing correctly
 * should handle email typing correctly
 * should handle password typing correctly
 * should call register function when register button is clicked
 * renders correctly
 * should have a link to login
 */
import React from 'react';
import {
  describe, it, expect, vi,
} from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import matchers from '@testing-library/jest-dom/matchers';
import { BrowserRouter } from 'react-router-dom';
import RegisterInput from './RegisterInput';

// Extend Vitest's expect with jest-dom matchers
expect.extend(matchers);

describe('RegisterInput component', () => {
  it('should handle name typing correctly', async () => {
    // Arrange
    render(
      <BrowserRouter>
        <RegisterInput register={() => {}} />
      </BrowserRouter>,
    );
    const nameInput = await screen.getByPlaceholderText('Name');

    // Action
    await userEvent.type(nameInput, 'John Doe');

    // Assert
    expect(nameInput).toHaveValue('John Doe');
  });

  it('should handle email typing correctly', async () => {
    // Arrange
    render(
      <BrowserRouter>
        <RegisterInput register={() => {}} />
      </BrowserRouter>,
    );
    const emailInput = await screen.getByPlaceholderText('Username');

    // Action
    await userEvent.type(emailInput, 'johndoe@example.com');

    // Assert
    expect(emailInput).toHaveValue('johndoe@example.com');
  });

  it('should handle password typing correctly', async () => {
    // Arrange
    render(
      <BrowserRouter>
        <RegisterInput register={() => {}} />
      </BrowserRouter>,
    );
    const passwordInput = await screen.getByPlaceholderText('Password');

    // Action
    await userEvent.type(passwordInput, 'password123');

    // Assert
    expect(passwordInput).toHaveValue('password123');
  });

  it('should call register function when register button is clicked', async () => {
    // Arrange
    const mockRegister = vi.fn();
    render(
      <BrowserRouter>
        <RegisterInput register={mockRegister} />
      </BrowserRouter>,
    );
    const nameInput = await screen.getByPlaceholderText('Name');
    await userEvent.type(nameInput, 'John Doe');
    const emailInput = await screen.getByPlaceholderText('Username');
    await userEvent.type(emailInput, 'johndoe@example.com');
    const passwordInput = await screen.getByPlaceholderText('Password');
    await userEvent.type(passwordInput, 'password123');
    const registerButton = await screen.getByRole('button', { name: 'Register' });

    // Action
    await userEvent.click(registerButton);

    // Assert
    expect(mockRegister).toHaveBeenCalledWith({
      name: 'John Doe',
      email: 'johndoe@example.com',
      password: 'password123',
    });
  });

  it('renders correctly', () => {
    // Arrange
    const { container } = render(
      <BrowserRouter>
        <RegisterInput register={() => {}} />
      </BrowserRouter>,
    );

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('should have a link to login', async () => {
    // Arrange
    render(
      <BrowserRouter>
        <RegisterInput register={() => {}} />
      </BrowserRouter>,
    );

    // Assert
    const loginLink = await screen.getByRole('link', { name: /login/i });
    expect(loginLink).toBeInTheDocument();
    expect(loginLink).toHaveAttribute('href', '/');
  });
});
