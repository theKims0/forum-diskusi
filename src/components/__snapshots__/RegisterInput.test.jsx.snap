// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegisterInput component > renders correctly 1`] = `
<div>
  <div
    class="login-content"
  >
    <form>
      <input
        class="form-input"
        placeholder="Name"
        type="text"
        value=""
      />
      <input
        class="form-input"
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        class="form-input"
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        class="button-primary"
        type="button"
      >
        Register
      </button>
    </form>
    <p>
      Already have an account?
       
      <a
        href="/"
      >
        Login
      </a>
    </p>
  </div>
</div>
`;
